tofu/imas2tofu/_core.py

```diff
@@ -2243,52 +2274,129 @@ def to_Config(self, Name=None, occ=None, indDescription=None, plot=True):
+                               + " limiter or mobile!")
+                        raise Exception(msg)
+                assert isinstance(mobile, bool)
+
+                # Get PFC
+                if mobile is True:
+                    units = wall.mobile.unit
+                else:
+                    units = wall.limiter.unit
+                nunits = len(units)
+
+                if nunits == 0:
+                    msg = ("There is no unit stored !\n"
+                           + "The required 2d description is empty:\n")
+                    ms = "len(idd.{}[occ={}].description_2d".format(ids, occ)
+                    msg += "{}[{}].limiter.unit) = 0".format(ms,
+                                                             description_2d)
+                    raise Exception(msg)
+
+                lS = [None for _ in units]
+                for ii in range(0, nunits):
+                    try:
+                        if mobile is True:
+                            outline = units[ii].outline[0]
+                        else:
+                            outline = units[ii].outline
+                        poly = np.array([outline.r, outline.z])
+
+                        if units[ii].phi_extensions.size > 0:
+                            pos, extent = units[ii].phi_extensions.T
+                        else:
+                            pos, extent = None, None
+                        name = units[ii].name
+                        cls, mobi = None, None
+                        if name == '':
+                            name = 'unit{:02.0f}'.format(ii)
+                        if '_' in name:
+                            ln = name.split('_')
+                            if len(ln) == 2:
+                                cls, name = ln
+                            elif len(ln) == 3:
+                                cls, name, mobi = ln
+                            else:
+                                name = name.replace('_', '')
+                        if cls is None:
+                            if ii == nunits-1:
+                                cls = 'Ves'
+                            else:
+                                cls = 'PFC'
+                        mobi = mobi == 'mobile'
+                        lS[ii] = getattr(mod, cls)(Poly=poly, pos=pos,
+                                                   extent=extent,
+                                                   Name=name, mobile=mobi,
+                                                   **kwargs)
+                    except Exception as err:
+                        msg = ("PFC unit[{}] named {} ".format(ii, name)
+                               + "could not be loaded!\n"
+                               + str(err))
+                        raise Exception(msg)
 
-            config = mod.Config(lStruct=lS, Name=Name, **kwargs)
+            if Name is None:
+                Name = wall.type.name
+                if Name == '':
```

Comment:
is this rightly indented ?

Reply:
yes, it happens that people save empty names ('') in IMAS, because they forget to fill it in